src/cpp/shared/game/save.cpp

```diff
@@ -900,9 +914,40 @@ void game::LoadGame(char* filnam, int newGame, int a3) {
       std::map<std::string, mapVariable> mapVariables;
       for (ironfist_save::save_t::mapVariable_const_iterator it = mp->mapVariable().begin();
         it != mp->mapVariable().end(); it++) {
-        mapVariable *mapVar = new mapVariable;
         std::string mapVariableId = it->id().get();
         MapVarType mapVariableType = StringToMapVarType(it->type());
+		if ((mapVariableType == MAPVAR_TYPE_BOOLEAN) && (!mapVariableId.empty())) {
+		  if ((mapVariableId[0] == '_') && ((it->value().get() == "true") || (it->value().get() == "True") || (it->value().get() == "TRUE"))) {
+			if (mapVariableId.length() > 10) {
+			  if (mapVariableId.substr(1, 8) == "AICHASE_") {
```

Comment:
I think this entire block can be replaced with a single sscanf call, or whatever its C++ equivalent is.

Reply:
If you can't simplify the code, then at least put it in a method. This is a great case of "complicated code that has simple behavior," which makes it a strong candidate for extracting a method.

Reply:
This has now been addressed.